core/src/main/scala/kafka/coordinator/group/CoordinatorLoaderImpl.scala

```diff
@@ -148,6 +156,16 @@ class CoordinatorLoaderImpl[T](
                 }
               }
 
+              val currentHighWatermark = log.highWatermark
+              if (currentHighWatermark > previousHighWatermark) {
+                onHighWatermarkUpdated.accept(currentHighWatermark)
+                previousHighWatermark = currentHighWatermark
+              }
+
+              if (currentOffset >= currentHighWatermark) {
+                onLoadedBatch.accept(currentOffset)
+              }
```

Comment:
`onLoadedBatch` updates the lastWrittenOffset so shouldn't we call this after updating `currentOffset`?

Reply:
currentOffset points to the nextOffset (the last offset in the batch plus 1) so I had this after successfully loading for an offset. Should lastWrittenOffset point to the next offset?

Reply:
you're right, we return
```
        // Required offset.
        partitionResult.lastOffset + 1
```
in CoordinatorPartitionWriter#append